common/Resources/ti.internal/extensions/node/fs.js

```diff
@@ -0,0 +1,1622 @@
+			this.birthtimeMs = this.birthtime.getTime();
+			this.ctimeMs = this.ctime.getTime();
+			this.mtimeMs = this.mtime.getTime();
+			this.size = this._file.size;
+			this.blocks = Math.ceil(this.size / this.blksize);
+			// TODO: Can we fake out the mode based on the readonly/writable/executable properties?
+		}
+	}
+
+	isFile() {
+		return this._file.isFile();
+	}
+
+	isDirectory() {
+		return this._file.isDirectory();
+	}
+
+	isBlockDevice() {
+		return false;
+	}
+
+	isCharacterDevice() {
+		return false;
+	}
+
+	isSymbolicLink() {
+		return this._file.symbolicLink;
+	}
+
+	isFIFO() {
+		return false;
+	}
+
+	isSocket() {
+		return false;
+	}
+}
+fs.Stats = Stats;
+
+class ReadStream {
+
+}
+fs.ReadStream = ReadStream;
+
+class WriteStream {
+
+}
+fs.WriteStream = WriteStream;
+
+/**
+ * @callback statsCallback
+ * @param {Error} err - Error if one occurred
+ * @param {fs.Stats} stats - file stats
+ */
+
+/**
+ * @param {string|URL|Buffer} path file path
+ * @param {integer} [mode=fs.constants.F_OK] accessibility mode/check
+ * @param {function} callback async callback
+ */
+fs.access = function (path, mode, callback) {
+	// cheat and just do a setTimeout?
+	setTimeout(() => {
+		if (typeof mode === 'function') {
+			callback = mode;
+			mode = fs.constants.F_OK;
+		}
+		try {
+			fs.accessSync(path, mode);
+		} catch (e) {
+			callback(e);
+			return;
+		}
+		callback();
+	}, 1);
+};
+
+/**
+ * @param {string|URL|Buffer} path file path
+ * @param {integer} [mode=fs.constants.F_OK] accessibility mode/check
+ */
+fs.accessSync = function (path, mode = fs.constants.F_OK) {
+	// F_OK is just whether file exists or not, no permissions check
+	// R_OK is read check
+	// W_OK is write check
+	// X_OK is execute check (acts like F_OK on Windows)
+	const fileHandle = getTiFileFromPathLikeValue(path);
+	if (!fileHandle.exists()) {
+		throw noSuchFile('access', path);
+	}
+
+	// TODO: We have no means of testing if a file is readable. It's assumed all files that exist under the app are?
+	if ((mode & fs.constants.W_OK) && !fileHandle.writable) {
+		throw permissionDenied('access', path);
+	}
+	if (!isWindows && (mode & fs.constants.X_OK) && !fileHandle.executable && fileHandle.isFile()) {
+		throw permissionDenied('access', path);
+	}
+};
+
+/**
+ * Asynchronously append data to a file, creating the file if it does not yet exist. data can be a string or a Buffer.
+ * @param {string|Buffer|URL|FileStream} file filepath to file
+ * @param {string|Buffer} data data to append to file
+ * @param {object|string} [options] options
+ * @param {string} [options.encoding='utf8'] encoding to use
+ * @param {integer} [options.mode=0o666] mode to create file, if not created
+ * @param {string} [options.flag='a'] file system flag
+ * @param {Function} callback function to call back with error if failed
+ */
+fs.appendFile = (file, data, options, callback) => {
+	callback = callback || options;
+	// TODO: Verify the callback is now a function!
```

Comment:
This needs to be done in most async function throughout the file. See node's [maybeCallback](https://github.com/nodejs/node/blob/53816cce699d02fb28a49b258e1fbc474568bbf8/lib/fs.js#L142) which does exactly this.